nemo/collections/asr/data/audio_to_text_dali.py

```diff
@@ -0,0 +1,404 @@
+        self._has_processed_signal = ('processed_signal' in out_dict and 'processed_signal_len' in out_dict)
+        if not self._has_processed_signal:
+            assert('audio' in out_dict and 'audio_len' in out_dict)
+        assert ('transcript' in out_dict and 'transcript_len' in out_dict)
+
+    def has_processed_signal(self):
+        return self._has_processed_signal
+
+    def get(self):
+        if self._has_processed_signal:
+            return self._outs['processed_signal'], self._outs['processed_signal_len'].reshape(-1), \
+                   self._outs['transcript'], self._outs['transcript_len'].reshape(-1)
+        else:
+            return self._outs['audio'], self._outs['audio_len'].reshape(-1), \
+                   self._outs['transcript'], self._outs['transcript_len'].reshape(-1)
+
+@experimental
+class AudioToCharDALIDataset(Iterator):
+    """
+    NVIDIA DALI pipeline that loads tensors via one or more manifest files where each line containing a sample descriptor in JSON,
+    including audio files, transcripts, and durations (in seconds).
+    Here's an example:
+    {"audio_filepath": "/path/to/audio.wav", "text_filepath": "/path/to/audio.txt", "duration": 23.147}
+    ...
+    {"audio_filepath": "/path/to/audio.wav", "text": "the transcription", "offset": 301.75, "duration": 0.82, "utt":
+    "utterance_id", "ctm_utt": "en_4156", "side": "A"}
+    Args:
+        manifest_filepath: Path to manifest file with the format described above. Can be comma-separated paths.
+        labels: String containing all the possible characters to map to.
+        sample_rate (int): Sample rate to resample loaded audio to.
+        batch_size (int): Number of samples in a batch.
+        num_threads (int): Number of CPU processing threads to be created by the DALI pipeline.
+        max_duration (float): Determines the maximum allowed duration, in seconds, of the loaded audio files.
+        min_duration (float): Determines the minimum allowed duration, in seconds, of the loaded audio files.
+        blank_index (int): blank character index, default = -1
+        unk_index (int): unk_character index, default = -1
+        normalize (bool): whether to normalize transcript text (default): True
+        bos_id (int): Id of beginning of sequence symbol to append if not None
+        eos_id (int): Id of end of sequence symbol to append if not None
+        trim (bool): If True, it will extract the nonsilent region of the loaded audio signal.
+        shuffle (bool): If set to True, the dataset will shuffled after loading.
+        device (str): Determines the device type to be used for preprocessing. Allowed values are: 'cpu', 'gpu'.
+        device_id (int): Index of the GPU to be used. Only applicable when device == 'gpu'. Defaults to 0.
+        global_rank (int): Worker rank, used for partitioning shards. Defaults to 0.
+        world_size (int): Total number of processes, used for partitioning shards. Defaults to 0.
```

Comment:
```suggestion
        world_size (int): Total number of GPUs participating in the training, used for partitioning shards. Defaults to 1.
```

Reply:
How about if we don't use GPUs?

Reply:
Right, maybe workers? I don't know what is the proper PyTorch nomenclature.